### DISCUSSION TITLE
How to get image URL

### DISCUSSION THREAD

**Discussion by ShayGus** [fuatakgun/eufy_security#814] [Q&A]
I would like to use WetRTC with generic camera.
The reason for this is that when streaming to devices the stream quality is good, but when streaming to the browser it's quite bad.
I would like to enjoy both worlds.
But defining generic camera requires both RTSP url and the image URL.
it seems that from the last version, the last image url isn't present any more.
So how can I get bot the stream and the last image url together?

**Comment by fuatakgun** [comment]
Smart move, makes sense. With latest version, the latest event picture is available as camera picture. Can you reference to another camera as a template maybe?

**Reply by ShayGus** [reply]
The settings for generic camera are stored in the HA settings, so you can't use templates there.
You need to provide a valid http url to get the image.

**Reply by fuatakgun** [reply]
Is it or you assume so?

https://www.home-assistant.io/integrations/generic/#still-image-url

**Reply by ShayGus** [reply]
You can't use it.
The thing is that to define generic camera, you need to start streaming.
At that point the entity goes from image to stream.
But to define generic camera, you need to provide, both, the static image and the stream and during the definition, the integration checks that both working.
When streaming the static image won't exist.

**Reply by fuatakgun** [reply]
if you use this integration (eufy_security), it will create camera entities, which you can use to get latest event photo.
via templating, you can use the camera image from this integration and create generic camera with still image with template.

**Reply by ShayGus** [reply]
How can I simultaneously get the stream and the still image using this integration?

**Reply by fuatakgun** [reply]
I have no more information to supply here, on the other hand, i feel like we are not able to move at any previous topics. You asked how to create a generic camera and i proposed to use templating, you disagreed, i pointed the documentation, the topic left open.

Later on, you asked how you can put a template value into it, i shared my idea, the topic left open.

So, good luck with your experiment and keep here updated with your findings please.

**Reply by fuatakgun** [reply]
Lastly, thanks to the latest updates from eufy, images are not publicly available and stored in an encrypted way and i am just rendering decrypted bytes. So, no more picture url.
